x-pack/plugins/upgrade_assistant/__jest__/client_integration/kibana_deprecations/deprecations_list.test.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { deprecationsServiceMock } from 'src/core/public/mocks';
+
+import { setupEnvironment } from '../helpers';
+import { KibanaTestBed, setupKibanaPage } from './kibana_deprecations.helpers';
+import { kibanaDeprecationsMockResponse } from './mocked_responses';
+
+const criticalDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.level === 'critical'
+);
+const warningDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.level === 'warning'
+);
+const configDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.deprecationType === 'config'
+);
+
+describe('Kibana deprecations table', () => {
+  let testBed: KibanaTestBed;
+  const { server } = setupEnvironment();
+  const deprecationService = deprecationsServiceMock.createStartContract();
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  beforeEach(async () => {
+    await act(async () => {
+      deprecationService.getAllDeprecations = jest
```

Comment:
Do you think it might make sense to add `setLoadKibanaDeprecations` functions to [`http_requests.ts`](https://github.com/elastic/kibana/blob/7.x/x-pack/plugins/upgrade_assistant/__jest__/client_integration/helpers/http_requests.ts#L15) that encapsulate this logic?

I understand the internal implementation would be inconsistent because for the other functions we're stubbing responses from the server, and in this case we'd use the deprecation service mock. But I think we could add a comment to explain why, and from the consumer's perspective the interface would be consistent, so maybe it's a nice DX improvement. Totally up to you.

Reply:
Thanks for the suggestion! I think that could be helpful. I'll look into this separately, as I think the overview tests could benefit from this as well.